src/components/SuiteHeader/SuiteHeader.jsx

```diff
@@ -106,6 +109,7 @@ const SuiteHeader = ({
   applications,
   sideNavProps,
   surveyLink,
+  onRequestLogout,
```

Comment:
I feel that the header itself should not be managing "links" but rather just call onLogout passing the current link that it knows about.  People can then override the onLogout function to do as they please.  This is true for changing applications as well.  ie, the navigation should call a function that can be overridden.  In graphite both logout and change application, show help, etc, all go through application-level APIs rather than hitting URL directly.  ie, there really should be a single standard uniform way in which the application header will leave an application for whatever reason.

Reply:
Sounds good @stuckless -- let's work on an interface that is nice for graphite to consume.

One possibility (to simplify the interface) would be to pass the header a single function prop `onRouteChange` that would be called (if defined) on every header link click event, with the type of action (logout, help, switch application, etc) and the URL. If the function returns true, the link redirect occurs in the header-- if false, it does not. This would give applications full control on how to handle (or fail) the route change.

Reply:
@bryancboyd @stuckless The `onRouteChange` function intercepts all route changes now. It is a `Promise<boolean>` that, if resolved with `true`, will allow the redirect to happen. If it resolves with `false`, it won't redirect. `onRouteChange` will be called with `(routeType, url, params)`. `routeType` is a string that can identify, for example, if the redirect is to a logout page, or to a documentation page. `url` is the actual URL to redirect to. The 3rd argument is only included when `routeType` is `APPLICATION`, so `params` will be `{appId: someapplicationid}`, for example